### PR TITLE
show math expressions in funman input panel

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

- Show math expressions for funman inputs
- Lower and upper bounds can be entered for the linearly constrained option within the math expression
- Fixed katex font

image

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue [hunk 1]
```diff
@@ -153,6 +178,30 @@ const variableOptions = computed(() => {
...
+			expressionPart = `d/dt ${variables[i]}`;
+		}
+		if (i === variables.length - 1) {
+			if (constraintType === ConstraintType.LessThan || constraintType === ConstraintType.Decreasing) {
+				expressionPart += `\\leq`;
+				expressionPart += constraintType === ConstraintType.Decreasing ? '0' : `${interval?.ub ?? 0}`;
+			} else {
+				expressionPart += `\\geq`;
+				expressionPart += constraintType === ConstraintType.Increasing ? '0' : `${interval?.lb ?? 0}`;
+			}
```
[the remark this answers is not in the document]
Comment (shawnyama) [reply packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue:197]: Ok, I'll put the equals to in if the closed upper bound is true. Even though we don't have designs to toggle that yet.

[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue:197]: No i mean there are other inconsistencies in general, even ignoring closed_upper_bound

- Funman has greater-than-or-equal-to, but the interface says greater-than, but your equation says greater-than or-equal-to
- Funman has, by default less than, and the interface says less-than, but the equation says less-than-or-equal to

File: packages/client/hmi-client/src/components/workflow/ops/funman/tera-funman-drilldown.vue [hunk 2]
```diff
@@ -41,9 +41,23 @@
 													<InputSwitch class="mr-3" v-model="knobs.compartmentalConstraint.isActive" />
 												</div>
 											</header>
-											<div class="section-row">
-												<katex-element :expression="stringToLatexExpression(`${stateIds.join(' + ')} = ${mass}`)" />
-												<span v-for="v of stateIds" :key="v"> {{ v }} &#8805; 0, </span>
+											<div class="flex align-items-center gap-6">
+												<div>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/funman/tera-funman-drilldown.vue:45]: Is this div necessary?

File: packages/client/hmi-client/src/components/workflow/ops/funman/tera-funman-drilldown.vue [hunk 3]
```diff
@@ -331,6 +352,7 @@ const runMakeQuery = async () => {
 			request.request.config.normalization_constant = parseFloat(mass.value);
 		}
 	}
+	console.log(request);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/funman/tera-funman-drilldown.vue:355]:
```suggestion
```

File: packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue [hunk 4]
```diff
@@ -153,6 +178,30 @@ const variableOptions = computed(() => {
...
+		let expressionPart = variables[i];
+		if (constraintType === ConstraintType.Increasing || constraintType === ConstraintType.Decreasing) {
+			expressionPart = `d/dt ${variables[i]}`;
+		}
+		if (i === variables.length - 1) {
+			if (constraintType === ConstraintType.LessThan || constraintType === ConstraintType.Decreasing) {
+				expressionPart += interval?.closed_upper_bound ? `\\leq` : `<`;
+				expressionPart += constraintType === ConstraintType.Decreasing ? '0' : `${interval?.ub ?? 0}`;
+			} else {
+				expressionPart += interval?.closed_upper_bound ? `\\geq` : `>`;
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue:195]: Should it be lower_bound here?

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue (hunk 4 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue:195]: Yeah I did this wrong actually even though there is no lower_bound

File: packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue [hunk 5]
```diff
@@ -117,7 +117,7 @@
 						@update:model-value="
 							($event) => {
 								const newWeights = cloneDeep(config.weights);
-								if (!newWeights) return;
+								if (isNaN($event)) return; // Don't accept empty value
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue]: You should put that test before creating `const newWeights`

File: packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue [hunk 6]
```diff
@@ -142,6 +142,7 @@
 				:expression="stringToLatexExpression(`\\forall \\ t \\in [${config.timepoints.lb}, ${config.timepoints.ub}]`)"
 			/>
 		</div>
+		{{ config.weights }}
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/workflow/ops/funman/tera-constraint-group-form.vue]:
```suggestion
```

**Review by YohannParis — APPROVED** [review]

**Pull request merged**